File: bdd/boilerplate/issue_171/go/main.go
package main

import (
	"context"
	"errors"
	"fmt"
	"net/http"
	"os"

	"github.com/gin-gonic/gin"
)

type MyRepro struct{}

func (s *MyRepro) FlattenAny(ctx context.Context, req *Issue171ReproServiceFlattenAnyRequest) (*Issue171ReproServiceFlattenAnyResponse, error) {
	m, ok := req.Payload.(map[string]any)
	if !ok || m["foo"] != "bar" {
		return nil, errors.New("invalid payload")
	}
	return &Issue171ReproServiceFlattenAnyResponse{}, nil
}
func (s *MyRepro) FlattenStructWithAny(ctx context.Context, req *Issue171ReproServiceFlattenStructWithAnyRequest) (*Issue171ReproServiceFlattenStructWithAnyResponse, error) {
	m, ok := req.Payload.Field.(map[string]any)
	if !ok || m["foo"] != "bar" {
		return nil, errors.New("invalid payload")
	}
	return &Issue171ReproServiceFlattenStructWithAnyResponse{}, nil
}
func main() {
	r := gin.Default()
	svc := &MyRepro{}
	RegisterIssue171ReproServiceHandler(r, svc)
	http.ListenAndServe(fmt.Sprintf(":%s", os.Getenv("PORT")), r)
}

File: bdd/boilerplate/issue_171/rust/src/main.rs
use async_trait::async_trait;
pub mod issue_171 {
    pub use crate::gen::issue_171::*;
}
mod gen {
    include!("../{{MODULE_NAME}}.rs");
}
struct MyRepro;
#[async_trait]
impl issue_171::ReproService for MyRepro {
    async fn flattenAny<'a>(&'a self, payload: xidl_rust_axum::serde_json::Value) -> Result<(), xidl_rust_axum::Error> {
        if payload.get("foo").and_then(|v| v.as_str()) == Some("bar") {
            Ok(())
        } else {
            Err(xidl_rust_axum::Error::bad_request())
        }
    }
    async fn flattenStructWithAny<'a>(&'a self, payload: issue_171::StructWithAny) -> Result<(), xidl_rust_axum::Error> {
        if payload.field.get("foo").and_then(|v| v.as_str()) == Some("bar") {
            Ok(())
        } else {
            Err(xidl_rust_axum::Error::bad_request())
        }
    }
}
#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let svc = issue_171::ReproServiceServer::new(MyRepro);
    let port = std::env::var("PORT").unwrap_or_else(|_| "8080".into());
    xidl_rust_axum::Server::builder().with_service(svc).serve(&format!("127.0.0.1:{}", port)).await?; Ok(())
}
